controllers/whoToFollow/whoToFollowV2.js

```diff
@@ -81,6 +82,10 @@ module.exports = async (req, res) => {
             ...userTopic,
             viewtype: 'user'
           });
+
+          if (!allUserIds.includes(userTopic.user_id)) {
+            allUserIds.push(userTopic.user_id);
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `Array.prototype.includes()` method has a time complexity of O(n). Since you're using it inside a loop, the overall time complexity becomes O(n^2), which can lead to performance issues when dealing with large arrays. Consider using a `Set` instead of an `Array` for `allUserIds` to improve performance as lookup in a Set is O(1).
```diff
- const allUserIds = [];
+ const allUserIds = new Set();

// And then, while adding elements
- if (!allUserIds.includes(userTopic.user_id)) {
-   allUserIds.push(userTopic.user_id);
- }
+ allUserIds.add(userTopic.user_id);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->